Source/Core/VideoCommon/Assets/ShaderAsset.cpp

```diff
@@ -45,26 +49,38 @@ bool ParseShaderProperties(const VideoCommon::CustomAssetLibrary::AssetID& asset
     std::transform(type.begin(), type.end(), type.begin(),
                    [](unsigned char c) { return std::tolower(c); });
 
-    if (type == "sampler2d")
-    {
-      property.m_type = ShaderProperty::Type::Type_Sampler2D;
-    }
-    else if (type == "samplercube")
-    {
-      property.m_type = ShaderProperty::Type::Type_SamplerCube;
-    }
-    else if (type == "samplerarrayshared_main")
-    {
-      property.m_type = ShaderProperty::Type::Type_SamplerArrayShared_Main;
-    }
-    else if (type == "samplerarrayshared_additional")
+    constexpr std::array<std::pair<std::string_view, ShaderProperty::Type>,
```

Comment:
This needs to be either `static` or outside the function (or both) otherwise it will construct this on the stack at runtime:
https://godbolt.org/z/sMP4r63hP
https://godbolt.org/z/e5oq7fjjK